src/main/java/com/mathworks/ci/TestFile.java

```diff
@@ -0,0 +1,94 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ */
+
+import java.util.List;
+import java.util.ArrayList;
+
+import org.apache.commons.lang.RandomStringUtils;
+
+public class TestFile {
+    private String filePath;
+    private String name;
+    private Double duration;
+    private String status;
+    private List<TestCase> testCases;
+    private String id;
+
+    public TestFile() {
+        filePath = "";
+        name = "";
+        duration = 0.0;
+        status = "NotRun";
+        testCases = new ArrayList<TestCase>();
+
+        id = RandomStringUtils.randomAlphanumeric(8);
+    }
+
+    public void incrementDuration(Double testCaseDuration) {
+        this.duration += testCaseDuration;
+    }
+
+    public void updateStatus(TestCase testCase) {
+        if (!status.equals("Failed")){
+            if (testCase.getFailed()){
+                status = "Failed";
+            }
+            else if (!status.equals("Incomplete")){
+                if (testCase.getIncomplete()){
+                    status = "Incomplete";
+                }
+                else if (testCase.getPassed()){
+                    status = "Passed";
+                }
+            }
+        }
+    }
+
+    public String getFilePath() {
+        return this.filePath;
+    }
+
+    public void setFilePath(String filePath) {

```

Comment:
I would refrain using `filePath` as name just to keep the class name differentiated as we heavily use FilePath class. variable may cause confusion.

Reply:
Yeah, I agree. Updated `filePath` to just `path`